src/DETHRACE/common/car.c

```diff
@@ -5593,161 +5584,153 @@ int CrashCarsTogetherSinglePass(br_scalar dt, int pPass, tCollison_data* collide
                             collide_list[i].car = car_2;
                             collided++;
                         }
-                    } else if (collide_list[i].car || collide_list[j].car) {
```

Comment:
I'm not sure how comfortable I feel with this large scale refactoring. On one hand the proposed code for sure looks more sensible, but I don't think I can look at the changes hard enough to know that we haven't introduced a subtle strange behavior. And rewriting the flow makes it very difficult to go back and step through following the OG code in a debugger. 

I actually don't know how to handle this. One idea is to wait until the single player game is (more or less) complete before cleaning the code up too much, and hopefully at that time we aren't comparing OG code? Do we break the PR into two? Everything above this line is easy to merge right now. What do you think?

Reply:
This is no refactoring, it's the same code with less indentations.

Reply:
But I'll remove the indentation patch to a separate pr.